### PR TITLE
Running WebApp auth on desktop.

### PR DESCRIPTION
**Pull request by RamjotSingh**
Allows applications declared in Web Application registrations to Acquire Token on desktop using ADAL. This allows same app to be used in Web as well as Desktop.

### CODE REVIEW

**Comment by vibronet** [comment]
Hi Ramjot,
thanks for the PR.
This scenario is not in scope for ADAL, and presents significant security challenges. The confidential client flow is reserved for pure browser based scenarios, where the code acquiring the tokens (and associated credentials required to perform the acquisition) lives on a server. Making client credentials available on a user-accessible machine would present challenges for key distribution and protection. Furthermore, there are a number of scenarios that apply to native clients only (like CA/device auth) and making it possible to call this method for a confidential client would create confusion and likely hard to troubleshoot issues.
In short, web applications should only be consumed via a browser, and native clients (no matter what technology they are implemented in, including Cordova/React native/Electron or any other stack using HTML/JS) should always be represented in Azure AD as public clients.
Per the above, we can't get this PR in. Thanks again!

**Comment by RamjotSingh** [comment]
Thanks for the clarification :)
